Resources/views/CategoryAdmin/tree.html.twig

```diff
@@ -23,7 +23,13 @@ file that was distributed with this source code.
             <li class="sonata-ba-list-field" objectId="{{ element.id }}" >
                 <div class="sonata-tree__item"{% if depth < 2 %} data-treeview-toggled{% endif %}>
                     {% if element.parent or root %}<i class="fa fa-caret-right" data-treeview-toggler></i>{% endif %}
-                    <a class="sonata-tree__item__edit" href="{{ admin.generateObjectUrl('edit', element) }}">{{ element.name }}</a>
+                    {% if admin.hasRoute('edit') and admin.hasAccess('edit') %}
+                        <a class="sonata-tree__item__edit" href="{{ admin.generateObjectUrl('edit', element) }}">{{ element.name }}</a>
+                    {% elseif admin.hasRoute('show') and admin.hasAccess('show') %}
+                        <a class="sonata-tree__item__edit" href="{{ admin.generateObjectUrl('show', element) }}">{{ element.name }}</a>
```

Comment:
wouldn't it make sense to simplify the code a bit like this:
```diff
- admin.hasRoute('show') and admin.hasAccess('show')
+ admin.availableAndAllowedRoute('show')
```

I currently have no idea of a good method name.... `availableAndAllowedRoute()` sounds bad 😄